web/src/stream_data.js

```diff
@@ -95,11 +95,12 @@ class BinaryDict {
     }
 }
 
-// The stream_info variable maps stream names to stream properties objects
+// The stream_info variable maps stream ids to stream properties objects
 // Call clear_subscriptions() to initialize it.
 let stream_info;
 
 const stream_ids_by_name = new FoldDict();
+const stream_ids_by_old_names = new FoldDict();
```

Comment:
This feels sort of ugly but I wasn't sure if there was a nicer way to do this while preserving the same functionality.

Reply:
Essentially `stream_ids_by_old_names` does what `stream_ids_by_name` used to do, and `stream_ids_by_name` allows us to still index `stream_info` by name efficiently when that's the only information we have.

Reply:
I think structurally the goal of this project is to eliminate things that need to access `stream_ids_by_old_names` at all, by just storing the stream ID wherever we are currently storing a stream name that could become stale, and doing all the lookups on that.

Reply:
Anyway this ugly thing seems fine; it makes the ugly thing we were doing more readable/explicit.

Reply:
Sounds good, I can keep iterating after this to use IDs in more places.

Reply:
See #26127 for further work. I haven't removed `stream_ids_by_old_names` yet because some of the filter/search logic uses stream names instead of IDs, and changing that (if we even wanted to, maybe behind the scenes) should probably be separate work.